ts/m4q.d.ts

```diff
@@ -1,328 +1,1140 @@
+
+        import(deep?: boolean): M4QObject<T>;
+
+        adopt(): M4QObject<T>;
+
+        remove(selector?: string): M4QObject<T>;
+
+        wrap(elements: Manipulatable): M4QObject<T>;
+
+        wrapAll(elements: Manipulatable): M4QObject<T>;
+
+        wrapInner(elements: Manipulatable): M4QObject<T>;
+
+        //endregion
+
+        //region src/position.js
+        offset(): Position;
+
+        offset(offset: Position): M4QObject<T>;
+
+        position(margin?: boolean): Position | undefined;
+
+        left(left: number): M4QObject<T>;
+
+        left(): number;
+
+        top(top: number): M4QObject<T>;
+
+        top(): number;
+
+        crood(): DOMRect | undefined;
+
+        pos(): Position | undefined;
+
+        //endregion
+
+        //region src/prop.js
+        prop(prop: string): any;
+
+        prop(prop: string, value: any): M4QObject<T>;
+
+        val(): string;
+
+        val(value: any): M4QObject<T>;
+
+        html(): string | undefined;
+
+        html(value: any): M4QObject<T>;
+
+        outerHtml(): string | undefined;
+
+        text(): string | undefined;
+
+        text(value: any): M4QObject<T>;
+
+        innerText(): string | undefined;
+
+        innerText(value: any): M4QObject<T>;
+
+        empty(): M4QObject<T>;
+
+        clear(): M4QObject<T>;
+
+        //endregion
+
+        //region src/script.js
+        script(): M4QObject<T>;
+
+        //endregion
+
+        //region src/size.js
+        height(): number;
+
+        height(value: number): M4QObject<T>;
+
+        width(): number;
+
+        width(value: number): M4QObject<T>;
+
+        outerHeight(): number;
+
+        outerHeight(value: number): M4QObject<T>;
+
+        outerWidth(): number;
+
+        outerWidth(value: number): M4QObject<T>;
+
+        padding(pseudo?: string): Rect;
+
+        margin(pseudo?: string): Rect;
+
+        border(pseudo?: string): Rect;
+
+        //endregion
+
+        //region src/utils.js
+        items(): T[];
+
+        //endregion
+
+        //region src/visibility.js
+        hide(): M4QObject<T>;
+
+        show(): M4QObject<T>;
+
+        hidden(hidden: boolean, callback?: (this: T, element?: T, hidden?: boolean, callback?: Function) => any): M4QObject<T>;
+
+        hidden(callback?: (this: T, element?: T, callback?: Function) => any): M4QObject<T>;
+
+        visible(mode: boolean, callback?: (this: T, element?: T, mode?: boolean, callback?: Function) => any): M4QObject<T>;
+
+        visible(callback?: (this: T, element?: T, callback?: Function) => any): M4QObject<T>;
+
+        toggle(callback?: (this: T, element?: T, callback?: Function) => any): M4QObject<T>;
+
+        //endregion
     }
-}
\ No newline at end of file
+}
+
+//region src/events.js
+interface Event {
+    isPropagationStopped: boolean;
+    isPreventedDefault: boolean;
+
+    stop(immediate?: boolean);
+}
+
+//endregion
+
+//region src/populate.js
+interface Window {
+    m4q: M4Q.M4Q;
```

Comment:
`m4q: M4Q;` // ??

Reply:
the M4Q represent static functions, this allow typescript can recognize window.m4q(...)